springwolf-plugins/springwolf-amqp-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/util/ProducerDataUtil.java

```diff
@@ -0,0 +1,41 @@
+package io.github.stavshamir.springwolf.asyncapi.util;
+
+import com.asyncapi.v2.binding.amqp.AMQPChannelBinding;
+import com.asyncapi.v2.binding.amqp.AMQPOperationBinding;
+import com.google.common.collect.ImmutableMap;
+import io.github.stavshamir.springwolf.asyncapi.types.ProducerData;
+import lombok.experimental.UtilityClass;
+
+import java.util.Collections;
+
+@UtilityClass
+public class ProducerDataUtil {
+    /**
+     * Constructs producer data with correctly set channel and operation bindings.
+     *
+     * @param channelName name of the channel
+     * @param exchange    name of the exchange
+     * @param routingKey  routing key
+     * @param payloadType type of the payload
+     * @return amqp producer data
+     */
+    public static ProducerData createAMQPProducerData(String channelName,
```

Comment:
Probably shouldn't be placed as a method on ProducerData since this is amqp-specific.